packages/cli/src/commands/composite/compile.ts

```diff
@@ -20,22 +23,43 @@ export default class CompositeCompile extends Command<Flags> {
       .map((token) => {
         return token.input
       })
-
-    if (allArgs.length < 2) {
-      this.spinner.fail('Missing composite path and at output path')
-      return
-    }
     try {
-      const [compositePath, ...outputPaths] = allArgs
-      const composite = await readEncodedComposite(this.ceramic, compositePath)
+      let composite: Composite | undefined = undefined
+      let outputPaths: Array<string> = []
+      if (this.stdin !== undefined && allArgs.length >= 1) {
+        const definition = JSON.parse(this.stdin) as EncodedCompositeDefinition
+        composite = await Composite.fromJSON({ ceramic: this.ceramic, definition })
+        outputPaths = allArgs
+      } else if (this.stdin === undefined && allArgs.length >= 2) {
+        composite = await readEncodedComposite(this.ceramic, allArgs[0])
+        outputPaths = allArgs.splice(1)
+      } else if (this.stdin !== undefined && allArgs.length < 1) {
+        this.spinner.fail(
+          'When the composite is passed as JSON in stdin, at least one output path needs to be given as param'
+        )
+        return
+      } else {
+        this.spinner.fail('Missing composite path and at output path')
+        return
+      }
       const runtimeDefinition = composite.toRuntime()
       outputPaths.map(async (outputPath) => {
         await writeRuntimeDefinition(runtimeDefinition, outputPath)
       })
-      this.spinner.succeed('Successfully saved compiled composite into given path(s)')
+      this.spinner.succeed('Compiling the composite... Done!')
+      if (!this.flags['disable-stdin']) {
+        outputPaths.forEach((path) => {
+          if (path.endsWith('.json')) {
+            // log the first .json path so that it can be piped e.g. to graphql:server or redirected to a file
+            this.log(path)
+            return
+          }
+        })
+      }
```

Comment:
If I correctly understand the intent here, the `return` statement won't do anything because it's part of the `forEach` callback, instead to ensure the command execution stops once one path is logged, we'd need something like:
```suggestion
        for (const path of outputPaths) {
          if (path.endsWith('.json')) {
            // log the first .json path so that it can be piped e.g. to graphql:server or redirected to a file
            this.log(path)
            return
          }
        }
      }
```